internal/hcp/internal/controllers/link/controller.go

```diff
@@ -52,19 +90,65 @@ func (r *linkReconciler) Reconcile(ctx context.Context, rt controller.Runtime, r
 		return err
 	}
 
+	// Validation - Ensure V2 Resource APIs are not enabled unless hcpAllowV2ResourceApis flag is set
 	var newStatus *pbresource.Status
 	if r.resourceApisEnabled && !r.hcpAllowV2ResourceApis {
 		newStatus = &pbresource.Status{
 			ObservedGeneration: res.Generation,
 			Conditions:         []*pbresource.Condition{ConditionDisabled},
 		}
-	} else {
-		newStatus = &pbresource.Status{
-			ObservedGeneration: res.Generation,
-			Conditions:         []*pbresource.Condition{ConditionLinked(link.ResourceId)},
+		if resource.EqualStatus(res.Status[StatusKey], newStatus, false) {
+			return nil
+		}
+		_, err = rt.Client.WriteStatus(ctx, &pbresource.WriteStatusRequest{
+			Id:     res.Id,
+			Key:    StatusKey,
+			Status: newStatus,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	hcpClient, err := r.hcpClientFn(&link)
+	if err != nil {
+		rt.Logger.Error("error creating HCP Client", "error", err)
+		return err
+	}
+
+	// Sync cluster data from HCP
+	cluster, err := hcpClient.GetCluster(ctx)
+	if err != nil {
+		rt.Logger.Error("error querying HCP for cluster", "error", err)
+		return r.linkingFailed(ctx, rt, res)
+	}
+
+	if link.HcpClusterUrl != cluster.HCPPortalURL {
```

Comment:
Can we also check and update the access mode here?

Reply:
done